src/platform/packages/shared/kbn-scout/src/playwright/fixtures/scope/worker/api_key.ts

```diff
@@ -77,63 +127,67 @@ export const requestAuthFixture = coreWorkerFixtures.extend<
-          role === 'admin'
+          predefinedRole === 'admin'
             ? {}
             : (() => {
-                const roleDescriptor = defaultRoles.get(role);
+                const roleDescriptor = defaultRoles.availableRoles.get(predefinedRole);
                 if (!roleDescriptor) {
-                  throw new Error(`Cannot create API key for non-existent role "${role}"`);
+                  const availableRoles = Array.from(defaultRoles.availableRoles.keys()).join(', ');
+                  const errorMessage = [
+                    `Role '${predefinedRole}' not found in the supported list: ${defaultRoles.rolesFilePath}. Available predefined roles: ${availableRoles}.`,
+                    `Is '${predefinedRole}' a custom test role? → Use getApiKeyForCustomRole() to log in with custom Kibana and Elasticsearch privileges`,
+                    `Is '${predefinedRole}' a predefined role? (e.g., admin, viewer, editor) → Add the role descriptor to ${defaultRoles.rolesFilePath} to enable it for testing.`,
+                  ].join('\n\n');
+                  throw new Error(errorMessage);
                 }
-                log.debug(
-                  `Creating API key for ${role} with privileges: ${JSON.stringify(roleDescriptor)}`
-                );
-                return { [role]: roleDescriptor };
+                return { [predefinedRole]: roleDescriptor };
               })();
+        return await createApiKeyWithAdminCredentials(predefinedRole, roleDescriptors);
+      };
 
-        const payload = createApiKeyPayload(apiKeyName, role, roleDescriptors);
-        const response = await apiClient.post('internal/security/api_key', {
-          headers: {
-            'kbn-xsrf': 'some-xsrf-token',
-            'x-elastic-internal-origin': 'kibana',
-            ...adminCookieHeader,
-          },
-          body: payload,
-          responseType: 'json',
-        });
-
-        if (response.statusCode !== 200) {
-          throw new Error(
-            `Failed to create API key for '${role}' role with response text: ${response.statusMessage}`
-          );
-        }
+      const getApiKeyForCustomRole = async (
+        roleDescriptor: KibanaRole | ElasticsearchRoleDescriptor
+      ): Promise<RoleApiCredentials> => {
+        isCustomRoleCreated = true;
 
-        const apiKey = response.body as ApiKey;
-        const apiKeyHeader = { Authorization: `ApiKey ${apiKey.encoded}` };
+        try {
+          await samlAuth.setCustomRole(roleDescriptor);
 
-        log.info(`Created API key for role: [${role}]`);
-        generatedApiKeys.push(apiKey);
-        return { apiKey, apiKeyHeader };
+          const result = await createApiKeyWithAdminCredentials(samlAuth.customRoleName, {
+            [samlAuth.customRoleName]: roleDescriptor,
+          });
+          return result;
+        } catch (error: any) {
+          log.error(`Failed to create API key for custom role: ${error.message}`);
+          throw error;
```

Comment:
_question_: isn't it duplicating the error handling in the `createApiKeyWithAdminCredentials` itself?

https://github.com/elastic/kibana/pull/243187/files#diff-6be56a3dcb9ff7e1ddfaae11bdc6ebac4dfc06f07595253773302c8799ad64e8R95-R99